### ISSUE TITLE
ValueError: The checkpoint you are trying to load has model type `zoedepth` but Transformers does not recognize this architecture

### ISSUE THREAD

**Issue by justinmanley** [huggingface/transformers#31477]
### System Info

- `transformers` version: 4.41.2 (also true of github rep head @ 4.42.0.dev0, installed via `pip install git+https://github.com/huggingface/transformers.git`)
- Platform: macOS-11.5.1-x86_64-i386-64bit
- Python version: 3.10.13
- Huggingface_hub version: 0.23.4
- Safetensors version: 0.4.2
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.2.1 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: ?? not sure

### Who can help?

@amyeroberts? (for vision -- depth estimation)

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Run the example code from https://huggingface.co/Intel/zoedepth-nyu-kitti:

```python
from transformers import pipeline

depth_estimator = pipeline(task="depth-estimation", model="Intel/zoedepth-nyu-kitti")
```

### Expected behavior

I expect the `depth_estimator` to be fully initialized without throwing any errors.

**Comment by amyeroberts** [comment]
Hi @justinmanley, 

ZoeDepth hasn't been merged into transformers yet -- #30136 -- and so can't be used with the pipeline. It should be ready to go soon! @NielsRogge is off for a bit, so I'd expect it to be available within maybe a week?

**Comment by justinmanley** [comment]
ahhh ok got it! that makes sense. thanks for explaining -- I'll keep an eye on https://github.com/huggingface/transformers/pull/30136!
